src/shaderlib/project/project.glsl.js

```diff
@@ -81,37 +75,42 @@ vec2 project_mercator_(vec2 lnglat) {
   );
 }
 
-vec2 project_position(vec2 position) {
-  if (projectionMode == PROJECT_IDENTITY) {
-    return position;
-  }
-  if (projectionMode == PROJECT_MERCATOR_OFFSETS) {
-    return project_scale(position);
+vec4 project_position(vec4 position) {
+
+  if (projectionMode == PROJECT_MERCATOR) {
+    return vec4(
+      project_mercator_(position.xy) * WORLD_SCALE * projectionScale,
+      project_scale(position.z),
+      position.w
+    );
   }
-  // Covers projectionMode == PROJECT_MERCATOR
-  return project_mercator_(position) * WORLD_SCALE * projectionScale;
+
+  // Apply model matrix
+  vec4 position_modelspace = modelMatrix * position;
```

Comment:
Yes. This is reasonable. The `modelMatrix` should be multiplied before `project_scale` so it's applied on top of original, untransformed, `position` in the "meters" world.